examples/memcached.md

```diff
@@ -0,0 +1,197 @@
+$ pod=$(kubectl get -n demo pods -l app.kubernetes.io/name=pmem-memcached -o jsonpath={..metadata.name})
+$ kubectl port-forward -n demo $pod 11211 &
+$ telnet 127.0.0.1 11211
+Trying 127.0.0.1...
+Connected to 127.0.0.1.
+
+set demo_key 0 500000 10
+I am PMEM.
+STORED
+get demo_key
+VALUE demo_key 0 10
+I am PMEM.
+END
+quit
+```
+
+This data really was stored in the data volume:
+```
+$ pod=$(kubectl get -n demo pods -l app.kubernetes.io/name=pmem-memcached -o jsonpath={..metadata.name})
+$ kubectl exec -n demo $pod grep 'I am PMEM.' /data/memcached-memory-file
+Binary file /data/memcached-memory-file matches
+```
+
+## Restartable Cache
+
+The [restartable cache
+feature](https://github.com/memcached/memcached/wiki/WarmRestart)
+works like non-persistent usage of PMEM. In addition, memcached writes
+out one additional, short state file during a shutdown triggered by
+`SIGUSR1`. The state file describes the content of the memory file and
+is used during a restart by memcached to decide whether it can use the
+existing cached data.
+
+The [example
+deployment](../deploy/kustomize/memcached/persistent/memcached-persistent.yaml)
+uses a stateful set because that can automatically create persistent
+volumes for each instance. The shell wrapper around memcached
+translates the normal [`SIGTERM` shutdown
+signal](https://kubernetes.io/docs/concepts/workloads/pods/pod/#termination-of-pods)
+into `SIGUSR1`.
+
+Deploying like that becomes less flexible because the memcached pods
+can no longer move freely between nodes. Instead, each pod has to be
+restarted on the node where its volume was created. There are also
+[several caveats for memcached in this
+mode](https://github.com/memcached/memcached/wiki/WarmRestart#caveats)
+that admins and application developers must be aware of.
+
+This example can also be kustomized. By default it uses the
+[`pmem-csi-sc-ext4` storage
+class](../deploy/common/pmem-storageclass-ext4.yaml), which must be
+installed in the cluster. Here we just use the defaults, in particular
+the default namespace:
+
+```
+$ kubectl apply -f deploy/kustomize/memcached/persistent/memcached-persistent.yaml
+service/pmem-memcached created
+statefulset.apps/pmem-memcached created
+```
+
+Beware that at the moment, the volumes need to be removed manually
+after removing the stateful set. A [request to automate
+that](https://github.com/kubernetes/kubernetes/issues/55045) has been
+pending for a while.
+
+```
+$ kubectl delete -f deploy/kustomize/memcached/persistent/memcached-persistent.yaml
+service "pmem-memcached" deleted
+statefulset.apps "pmem-memcached" deleted
+$ kubectl delete pvc -l app.kubernetes.io/name=pmem-memcached
+persistentvolumeclaim "memcached-data-volume-pmem-memcached-0" deleted
+```
+
+We can verify that memcached really does a warm restart by storing some
+data, removing the instance and then starting it again.
+```
+$ kubectl apply -f deploy/kustomize/memcached/persistent/memcached-persistent.yaml
```

Comment:
I'd really consider breaking up this long code block into separate more manageable code blocks.  I found this a bit difficult to follow especially when telnet commands and output are intermixed.